drivers/pwm/pwm_mcux_sctimer.c

```diff
@@ -76,7 +76,8 @@ static int mcux_sctimer_pwm_set_cycles(const struct device *dev,
 		return 0;
 	}
 
-	if (period_cycles != data->period_cycles[channel]) {
+	if (period_cycles != data->period_cycles[channel] &&
+	    duty_cycle != data->channel[channel].dutyCyclePercent) {
```

Comment:
could you check for pulse cycles here instead and have one less struct field access and achieve the same effect?

Reply:
on second glance, I guess period cycles is stored in the data struct but not pulse cycles...